src/proofs.rs

```diff
@@ -0,0 +1,983 @@
+use crate::*;
+
+use crate::desugar::{desugar_commands, literal_name, Desugar};
+use crate::typechecking::FuncType;
+use symbolic_expressions::Sexp;
+
+pub const RULE_PROOF_KEYWORD: &str = "rule-proof";
```

Comment:
This keyword allows you to get the proof for a rule firing in the actions of a rule.